R/use_data_dir.R

```
@@ -0,0 +1,44 @@
+#' Make files in a directory accessible to tutorials
+#'
+#' By default, `learnr` tutorials can access files in a `data/` directory within
+#' the same directory as the tutorial's R Markdown file. `use_data_dir()` allows
+#' you to expose files in a different directory (e.g. the `data/` directory of a
+#' different tutorial) to avoid duplicating files.
+#'
+#' @param dir A directory
+#'
+#' @return Invisibly returns the specified directory
+#' @export
+#' @examples \dontrun{
+#' use_data_dir("../other_tutorial/data")
+#' }
+
+use_data_dir <- function(dir = "data") {
+  if (!dir.exists(dir)) {
+    rlang::warn(paste0('The data directory "', dir, '" could not be found.'))
+  }
+
+  options(learnr.data_dir = dir)
+
+  invisible(dir)
+}
+
+copy_data_dir <- function(exercise_dir) {
+  # First check environment variable, then options(), then default to "data/"
+  source_dir <- Sys.getenv(
+    "LEARNR_DATA_DIR", unset = getOption("learnr.data_dir", default = "data")
+  )
+
+  if (!dir.exists(source_dir)) {return(invisible(NULL))}
```

We should report this failure, probably with a warning

---

I can see why it's currently invisible -- if there isn't a `data/` directory we don't want to add a lot of noise. So we should put some thought into how we can surface warnings when its clear that the user would want to see them without creating  noise for tutorials without data folders.

---

I've changed this to return invisibly if "data/" doesn't exist, but to raise an error if a different directory is specified (by envvar or `options()`) and that directory doesn't exist (5b4a0ee).